File: Competitions/aces_pre_coders_v9/BiggestIsland.py
def is_valid(x, y, n, m):
    return 0 <= x < n and 0 <= y < m

def dfs(grid, x, y, n, m):
    if not is_valid(x, y, n, m) or grid[x][y] == 0:
        return 0

    grid[x][y] = 0  # Mark the cell as visited
    area = 1  # Initialize area for this island

    # 8-adjacent directions
    directions = [(1, 0), (-1, 0), (0, 1), (0, -1), (1, 1), (-1, -1), (1, -1), (-1, 1)]

    for dx, dy in directions:
        new_x, new_y = x + dx, y + dy
        area += dfs(grid, new_x, new_y, n, m)

    return area

def find_largest_island(grid):
    n = len(grid)
    m = len(grid[0])
    max_area = 0

    for i in range(n):
        for j in range(m):
            if grid[i][j] == 1:
                island_area = dfs(grid, i, j, n, m)
                max_area = max(max_area, island_area)

    return max_area

# Input parsing
n = int(input())
m = int(input())

grid = []
for _ in range(n):
    row = list(map(int, input().strip().split()))
    grid.append(row)

# Find and output the area of the largest island
largest_island_area = find_largest_island(grid)
print(largest_island_area)

File: Competitions/Aces/aces_pre_coders_v9/Biggest Island.cpp
#include <bits/stdc++.h>

using namespace std;
#define MAX 10
int grid[MAX][MAX];
bool visited[MAX][MAX];
int row, col;
int c;
int x_new, y_new;
int t;

int x_walk[] = {0, 0, 1, -1, 1, -1, 1, -1};
int y_walk[] = {1, -1, 0, 0, 1, -1, -1, 1};

bool isValid(int x, int y)
{
    if (x < 0 || y < 0 || x >= row || y >= col || grid[x][y] == 0)
        return false;

    return true;
}

void dfs(int x, int y)
{

    if (visited[x][y])
        return;

    visited[x][y] = true;
    t++;

    for (int i = 0; i < 8; i++)
    {
        x_new = x + x_walk[i];
        y_new = y + y_walk[i];

        if (isValid(x_new, y_new))
            dfs(x_new, y_new);
    }
}

int main()
{
    cin >> row;
    cin >> col;

    for (int i = 0; i < row; i++)
    {
        for (int j = 0; j < col; j++)
        {
            cin >> c;
            grid[i][j] = c;
        }
    }

    int max = 0;
    for (int i = 0; i < row; i++)
    {
        for (int j = 0; j < col; j++)
        {
            t = 0;
            if (!visited[i][j] && grid[i][j] == 1)
            {
                dfs(i, j);
                if (t > max)
                    max = t;
            }
        }
    }

    cout << max << endl;

    return 0;
}